src/typing.py

```diff
@@ -1916,7 +1916,21 @@ def __new__(cls, typename, bases, ns):
             raise TypeError("Class syntax for NamedTuple is only supported"
                             " in Python 3.6+")
         types = ns.get('__annotations__', {})
-        return _make_nmtuple(typename, types.items())
+        nm_tpl = _make_nmtuple(typename, types.items())
+        saw_default = False
```

Comment:
I think ``saw_default`` is redundant. You could use ``elif defaults:`` or ``elif defaults_dict:``, instead of ``elif saw_default:`` below.